update documents related to Plan Cache (#7956)
pingcap/docs: sql-prepare-plan-cache.md

+------------------------+
1 row in set (0.00 sec)
```

> ## Manually clear execution plan cache

Author: @qw4990 Should we remove 'manually' since there is no where describing 'automatically clear xxx'? Seems this info is unnecessary.

You can manually clear execution plan cache by executing the `ADMIN FLUSH [SESSION | INSTANCE] PLAN_CACHE` statement.

> In this statement, `[SESSION | INSTANCE]`specifies whether the plan cache is cleared for the current session or the whole TiDB instance. If the scope is left blank, the statement above applies to `SESSION` cache by default.

Reviewer: ```suggestion
In this statement, `[SESSION | INSTANCE]`specifies whether the plan cache is cleared for the current session or the whole TiDB instance. If the scope is not specified, the statement above applies to the `SESSION` cache by default.
```

Empty set (0.00 sec)

MySQL [test]> select @@last_plan_from_cache; -- Select the cached plan
+------------------------+
| @@last_plan_from_cache |
+------------------------+
|                      1 |
+------------------------+
1 row in set (0.00 sec)

MySQL [test]> admin flush session plan_cache; -- Clear the cached plan of the current session
Query OK, 0 rows affected (0.00 sec)

MySQL [test]> execute stmt;
Empty set (0.00 sec)

MySQL [test]> select @@last_plan_from_cache; -- The cached plan cannot be selected again, because it has been cleared
+------------------------+
| @@last_plan_from_cache |
+------------------------+
|                      0 |
+------------------------+
1 row in set (0.00 sec)
```

Currently, TiDB does not support clearing `GLOBAL` execution plan cache. That means you cannot clear the cached plan of the whole TiDB cluster. The following error is reported if you try to clear the `GLOBAL` execution plan cache:

{{< copyable "sql" >}}

```sql
MySQL [test]> admin flush global plan_cache;
ERROR 1105 (HY000): Do not support the 'admin flush global scope.'
```

## Ignore the `COM_STMT_CLOSE` command and the `DEALLOCATE PREPARE` statement

> To reduce syntax analysis for executed SQL statements, it is recommended that you run `prepare stmt` once, then `execute stmt` multiple times before running `deallocate prepare`:

Reviewer 2: `syntax analysis` --> `optimization cost`
`Prepare/Execute` can help skip the whole optimization phase, and `syntax analysis` is just a part of optimization, so using `optimization cost` here seems more precise.

> To reduce syntax analysis for executed SQL statements, it is recommended that you run `prepare stmt` once, then `execute stmt` multiple times before running `deallocate prepare`:

Author: ```suggestion
To reduce optimization cost for executed SQL statements, it is recommended that you run `prepare stmt` once, then `execute stmt` multiple times before running `deallocate prepare`:
```

> To reduce optimization cost for executed SQL statements, it is recommended that you run `prepare stmt` once, then `execute stmt` multiple times before running `deallocate prepare`:

Reviewer: ```suggestion
To reduce the syntax parsing cost of SQL statements, it is recommended that you run `prepare stmt` once, then `execute stmt` multiple times before running `deallocate prepare`:
```

{{< copyable "sql" >}}

```sql
MySQL [test]> prepare stmt from '...'; -- prepare once
MySQL [test]> execute stmt using ...;  -- execute once
MySQL [test]> ...
MySQL [test]> execute stmt using ...;  -- execute multiple times
MySQL [test]> deallocate prepare stmt; -- Release the prepared statement
```

In real practice, you may be used to running `deallocate prepare` each time after running `execute stmt`, as shown below:

{{< copyable "sql" >}}

```sql
MySQL [test]> prepare stmt from '...'; -- First prepare
MySQL [test]> execute stmt using ...;
MySQL [test]> deallocate prepare stmt; -- Release the prepared statement
MySQL [test]> prepare stmt from '...'; -- Second prepare
MySQL [test]> execute stmt using ...;
MySQL [test]> deallocate prepare stmt; -- Release the prepared statement
```

> In such practice, the plan obtained by the first execute statement cannot be reused by the second execute statement.

Reviewer: ```suggestion
In such practice, the plan obtained by the first executed statement cannot be reused by the second executed statement.
```

> To address the problem, you can set the system varible [`tidb_ignore_prepared_cache_close_stmt`](/system-variables.md#tidb_ignore_prepared_cache_close_stmt-new-in v60) to `ON` so TiDB ignores commands to close `prepare stmt`:

Reviewer: ```suggestion
To address the problem, you can set the system varible [`tidb_ignore_prepared_cache_close_stmt`](/system-variables.md#tidb_ignore_prepared_cache_close_stmt-new-in-v600) to `ON` so TiDB ignores commands to close `prepare stmt`:
```

> To address the problem, you can use the system varible [`tidb_ignore_prepared_cache_close_stmt`](/system-variables.md#tidb_ignore_prepared_cache_close_stmt-new-in v60). After this variable is set to `ON`, TiDB ignores commands to close `prepare stmt`:

Reviewer: ```suggestion
To address the problem, you can set the system varible [`tidb_ignore_prepared_cache_close_stmt`](/system-variables.md#tidb_ignore_prepared_cache_close_stmt-new-in v60) to `ON` so TiDB ignores commands to close `prepare stmt`:
```
